src/shared/shortcut.js

```diff
@@ -0,0 +1,113 @@
+import { useEffect } from 'preact/hooks';
+
+// Bit flags indicating modifiers required by a shortcut or pressed in a key event.
+const modifiers = {
+  alt: 1,
+  ctrl: 2,
+  meta: 4,
+  shift: 8,
+};
+
+/**
+ * Match a `shortcut` key sequence against a keydown event.
+ *
+ * A shortcut key sequence is a string of "+"-separated keyboard modifiers and
+ * keys. The list may contain zero or more modifiers and must contain exactly
+ * one non-modifier key. The key and modifier names are case-insensitive.
+ * For example "Ctrl+Enter", "shift+a". Key names are matched against `KeyboardEvent.key`.
+ *
+ * @param {KeyboardEvent} event
+ * @param {string} shortcut
+ * @return {boolean}
+ */
+export function matchShortcut(event, shortcut) {
```

Comment:
matchShortcut(createEvent('a', { ctrl: true }), 'ctrl+a+b'));  returns `false`
matchShortcut(createEvent('a', { ctrl: true }), 'ctrl+b+a'));  returns `true`

Do you see an issue with mal formatted strings like these two? One idea is to throw an exception if more than 1 key is detected.

Reply:
Specifying multiple keys in the call to `matchShortcut` would be incorrect usage, so throwing an exception makes sense to me.